src/views/Workflow.vue

```diff
@@ -16,44 +16,51 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 -->
 
 <template>
-  <div class="workflow-panel fill-height">
-    <lumino
-      ref="lumino"
-      v-on:lumino:deleted="onWidgetDeletedEvent"
-      tab-title-prop="tab-title"
-    >
-      <v-skeleton-loader
-        v-for="widgetId of treeWidgets"
-        :key="widgetId"
-        :id="widgetId"
-        :loading="isLoading"
-        type="list-item-three-line"
-        tab-title="tree"
+  <div>
+    <toolbar
+      v-on:add-tree="this.addTreeWidget"
+      v-on:add-graph="this.addGraphWidget"
+      v-on:add-mutations="this.addMutationsWidget"
```

Comment:
This is basically what replaces the Event Bus.

Instead of having the `Toolbar` component emitting global events, we give it a listeners for each custom event (`add-tree`, `add-graph`, and `add-mutations`).

This way `Toolbar` is able to use those listeners as `v-on:click=$listeners['add-graph]`, effectively sending an event to the listener of `add-graph` (this `Workflow` view in this case).

Further down, this view takes care to add a tree widget by communicating with the tree widget.

No global events, no event bus, no need to worry about cleaning up event listeners, and easier to understand how components are linked (good practice in Vue to send data top down this way :+1: )